src/main/java/org/myrobotlab/programab/handlers/oob/OobProcessor.java

```diff
@@ -0,0 +1,93 @@
+package org.myrobotlab.programab.handlers.oob;
+
+import java.util.List;
+
+import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.Message;
+import org.myrobotlab.programab.models.Mrl;
+import org.myrobotlab.programab.models.Oob;
+import org.myrobotlab.service.ProgramAB;
+
+public class OobProcessor {
+
+  private static OobProcessor instance;
+  private transient ProgramAB programab;
+  protected int maxBlockTime = 2000;
+
+  private OobProcessor() {
+  }
+
+  public static OobProcessor getInstance(ProgramAB programab) {
```

Comment:
Good point .. there might be problems... not sure I'll needs some tests.
The reason its a singleton is because it needed in a session and the sriax handler.
I was trying to limit resources of xml parser creation.
Probably a simple way is to have this part of ProgramAB, since both the sessions and the handler have references to this.

Reply:
removed over optimizing, no longer a singleton